src/server/scripts/Northrend/VaultOfArchavon/boss_archavon.cpp

```diff
@@ -107,15 +107,17 @@ class boss_archavon : public CreatureScript
                             events.ScheduleEvent(EVENT_ROCK_SHARDS, 15000);
                             break;
                         case EVENT_CHOKING_CLOUD:
-                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, [&](Unit* u) { return u && !(u->GetDistance(me->GetPosition()) < 10.0f); }))
```

Comment:
read `UnitAI.h` 159-166

Reply:
SELECT_TARGET_FARTHEST would make the script to always target the one, truly, farthest away instead of randomly selecting someone.

Atleast that's the results after trying it with the enum

Reply:
Who said you should use `SELECT_TARGET_FARTHEST`?

[UnitAI.h L162](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/CoreAI/UnitAI.h#L159-L166)

Reply:
SELECT_TARGET_FARTHEST implies the farthest target will be selected, no? That's not the intended behavior, it should just select a random ranged target

Reply:
> Who said you should use `SELECT_TARGET_FARTHEST`?

Reply:
```if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1,  -10.0f))``` If I understood it correctly

Reply:
yup (assuming he can't jump on MT, idk if that's correct)

Reply:
MT is always melee range so yes.
I guess he should be possible to do it if MT runs away but yeah. 

Thanks for showing this by the way, made it simpler.